### DISCUSSION TITLE
Allow Mod Presets

### DISCUSSION THREAD

**Discussion by Githubberclown** [FakeFishGames/Barotrauma#6912] [Ideas and suggestions]
Feature Request: Add Mod Presets That You Can Create So You Don't Have To Enable Every Mod Before You Start Your Server

**Comment by StruggleDuck** [comment]
This is would be awesome. it would make operating a server way easier, and would definitely increase quality of life. on top of that, it shouldn't even be that difficult to add.

**Comment by SpazzMaticus107** [comment]
It would be super helpful if you play with more than one crew, build subs or mod. +1

**Comment by Ek-Videogames** [comment]
+1
Only took Garry's-Mod 7 years to figure out how.

A [ DISABLE ALL ] button would also be good so you don't have to click them one at a time and wait 30 seconds for the entire game to recache.

**Comment by StruggleDuck** [comment]
I also believe that adding a simple tag beside each mod which tells you what kind of mod it is based off of the steam workshop tags would be awesome. It is super annoying to be scrolling through my massive list of mods, and trying to guess whether or not a mod is a custom submarine, or a mod that adds a bunch of random shit that I dont want.

**Reply by ghost** [reply]
sort by tags button might be pog

**Comment by pavel4win** [comment]
Although mod presets already realized for windows game client, I have realized a mod preset for a dedicated server.

I have realized mod presets with windows cmd and linux bash scripts. If anyone interested. I can publish my scripts here.
my scripts allow:
1. to start a Barotrauma dedicated server with mods preset (just by replacing config_player.xml and serversettings.xml
2. make a backup game save in the end of a day in the folder with datetime name.
3. automatically update dedicated server and all mods including Lua
4. upload saves to linux dedicated sever through Putty SSH client and/or download it back to win client.
5. To filter a dedicated server logs in real time by keywords, player name, etc. Implemented both in windows and in linux.

**Comment by pavel4win** [comment]
> Feature Request: Add Mod Presets That You Can Create So You Don't Have To Enable Every Mod Before You Start Your Server

![image](https://github.com/user-attachments/assets/6650b9ae-ccc1-41f6-a879-7053d5a76942)
